File: 217.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        num_set = set()
        for i in range(0, len(nums)):
            if nums[i] in num_set:
                return True
            else:
                num_set.add(nums[i])
        
        return False


if __name__ == '__main__':
    test_cases = ( ([1,2,3,1], True), ([1,2,3,4], False), ([1,1,1,3,3,4,3,2,4,2], True))
    for case in test_cases:
        print(f'Input: {case[0]}\nExpected Output: {case[1]}\nActual output: {Solution().containsDuplicate(case[0])}\n')

File: 217.ts
function containsDuplicate(nums: number[]): boolean {
    let numSet = new Set<number>;
    for(const num of nums)
    {
        if(numSet.has(num))
        {
            return true;
        }
        else
        {
            numSet.add(num);
        }
    }
    return false;
};

let testCases217: Array<Array<any>> = [ [[1,2,3,1], true], [[1,2,3,4], false], [[1,1,1,3,3,4,3,2,4,2], true]];
for(const testCase of testCases217)
{
    console.log(`
        Input: ${testCase[0]}
        Expected output: ${testCase[1]}
        Actual output: ${containsDuplicate(testCase[0])}
    `)
}